src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -381,6 +383,135 @@ public Task GetMigratedServerUrls()
+        public async Task MigrateFromAnotherDatabase()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var blittable = await context.ReadForMemoryAsync(RequestBodyStream(), "migration-configuration");
+                var migrationConfiguration = JsonDeserializationServer.MigrationConfiguration(blittable);
+
+                if (string.IsNullOrWhiteSpace(migrationConfiguration.DatabaseTypeName))
+                    throw new ArgumentException("DatabaseTypeName cannot be null or empty");
+
+                if (string.IsNullOrWhiteSpace(migrationConfiguration.FullPathToMigrator))
+                    throw new ArgumentException("FullPathToMigrator cannot be null or empty");
+
+                if (migrationConfiguration.Input == null)
+                    throw new ArgumentException("Input cannot be null");
+
+                if (Directory.Exists(migrationConfiguration.FullPathToMigrator) == false)
+                    throw new InvalidOperationException($"Directory {migrationConfiguration.FullPathToMigrator} doesn't exist");
+
+                const string migratorFileName = "Raven.Migrator.dll";
+                var path = Path.Combine(migrationConfiguration.FullPathToMigrator, migratorFileName);
+                if (File.Exists(path) == false)
+                    throw new InvalidOperationException($"The file '{migratorFileName}' doesn't exist in path: {migrationConfiguration.FullPathToMigrator}");
+
+                var processStartInfo = new ProcessStartInfo
+                {
+                    FileName = "dotnet",
+                    Arguments = $"{migratorFileName} {migrationConfiguration.DatabaseTypeName}",
+                    WorkingDirectory = migrationConfiguration.FullPathToMigrator,
+                    CreateNoWindow = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    RedirectStandardInput = true,
+                    UseShellExecute = false
+                };
+
+                Process process = null;
+                try
+                {
+                    process = Process.Start(processStartInfo);
+                }
+                catch (Exception e)
+                {
+                    process?.Kill();
+                    throw new InvalidOperationException("Unable to execute Migrator." + Environment.NewLine +
+                                                        "Command was: " + Environment.NewLine +
+                                                        (processStartInfo.WorkingDirectory ?? Directory.GetCurrentDirectory()) + "> "
+                                                        + processStartInfo.FileName + " " + processStartInfo.Arguments, e);
+                }
+
+                await process.StandardInput.WriteLineAsync(migrationConfiguration.Input.ToString());
```

Comment:
what do we pass here? is it  "Command - available commands: databases, collections, export"?

Reply:
https://github.com/grisha-kotler/ravendb/blob/fd50c270b7f01149d8cd7d13af5e738c6be100b9/tools/Raven.Migrator/MongoDB/MongoDBConfiguration.cs

or

https://github.com/grisha-kotler/ravendb/blob/fd50c270b7f01149d8cd7d13af5e738c6be100b9/tools/Raven.Migrator/CosmosDB/CosmosDBConfiguration.cs#L3

Reply:
It can be any migration configuration

Reply:
Thanks for clarification. `InputConfiguration` might be better property name then (minor)

Reply:
done